src/analyses/base.ml

```diff
@@ -805,18 +792,11 @@ struct
     | TPtr _ -> `Address AD.top_ptr
     | TComp ({cstruct=true} as ci,_) -> `Struct (top_comp ci)
     | TComp ({cstruct=false},_) -> `Union (ValueDomain.Unions.top ())
-    | TArray (ai, exp, _) ->
-      let default = `Array (ValueDomain.CArrays.top ()) in
-      (match exp with
-       | Some exp ->
-         (match eval_rv a gs st exp with
-          | `Int n -> begin
-              match ID.to_int n with
-              | Some n -> `Array (ValueDomain.CArrays.make (Int64.to_int n) (if get_bool "exp.partition-arrays.enabled" then (top_value a gs st ai) else (bot_value a gs st ai)))
-              | _ -> default
-            end
-          | _ -> default)
-       | None -> default)
+    | TArray (ai, None, _) ->
+      `Array (ValueDomain.CArrays.make None (if get_bool "exp.partition-arrays.enabled" then (top_value a gs st ai) else (bot_value a gs st ai)))
+    | TArray (ai, Some exp, _) ->
+      let l = Cil.isInteger (Cil.constFold true exp) in
```

Comment:
Why replace `eval_rv` when `Cil.constFold` does strictly less?

Reply:
See above

Reply:
For a VLA with some known length, it might be that we overwrite the length here. Probably length should also be a lattice (interval), then there's no problem for VLAs/malloced arrays.

Reply:
Changed it so the length is from `Idx` (domain also used for indices).